homeassistant/components/luxtronik/sensor.py

```diff
@@ -0,0 +1,613 @@
+"""Luxtronik heatpump sensor."""
+# region Imports
+from typing import Final
+
+from homeassistant.components.sensor import (
+    ENTITY_ID_FORMAT,
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_FRIENDLY_NAME,
+    CONF_ICON,
+    CONF_ID,
+    CONF_SENSORS,
+    DEVICE_CLASS_TEMPERATURE,
+    ENERGY_KILO_WATT_HOUR,
+    ENTITY_CATEGORY_DIAGNOSTIC,
+    EVENT_HOMEASSISTANT_STOP,
+    STATE_UNAVAILABLE,
+    TEMP_CELSIUS,
+    TIME_HOURS,
+    TIME_SECONDS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import ENTITY_CATEGORIES, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import (
+    ATTR_STATUS_TEXT,
+    CONF_GROUP,
+    CONF_LANGUAGE_SENSOR_NAMES,
+    DEFAULT_DEVICE_CLASS,
+    DEVICE_CLASSES,
+    DOMAIN,
+    ICONS,
+    LOGGER,
+    LUX_SENSOR_STATUS,
+    LUX_SENSOR_STATUS1,
+    LUX_SENSOR_STATUS3,
+    LUX_STATE_ICON_MAP,
+    LUX_STATES_ON,
+    LUX_STATUS1_WORKAROUND,
+    LUX_STATUS3_WORKAROUND,
+    LUX_STATUS_HEATING,
+    LUX_STATUS_NO_REQUEST,
+    PLATFORMS,
+    UNITS,
+)
+from .helpers.helper import get_sensor_text, get_sensor_value_text
+from .luxtronik_device import LuxtronikDevice
+from .model import LuxtronikStatusExtraAttributes
+
+# endregion Imports
+
+# region Constants
+SECOUND_TO_HOUR_FACTOR: Final = 0.000277777777778
+# endregion Constants
+
+# region Setup
+
+
+async def async_setup_platform(
```

Comment:
This integration should not support YAML and thus does not need a platform setup.